DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageFunctionUtil.java

```diff
@@ -19289,6 +19290,121 @@ public static short shortCast(Object a) {
+        }
+        PyObject o = (PyObject) a;
+        if (o == null || o.isNone()) {
+            return QueryConstants.NULL_INT;
+        }
+        return o.getIntValue();
+    }
+
+    public static double doublePyCast(Object a) {
+        if (a != null && !(a instanceof PyObject)) {
+            throw new IllegalArgumentException("provided value is not a PyObject");
+        }
+        PyObject o = (PyObject) a;
+        if (o == null || o.isNone()) {
+            return QueryConstants.NULL_DOUBLE;
+        }
+        return o.getDoubleValue();
+    }
+
+    public static long longPyCast(Object a) {
+        if (a != null && !(a instanceof PyObject)) {
+            throw new IllegalArgumentException("provided value is not a PyObject");
+        }
+        PyObject o = (PyObject) a;
+        if (o == null || o.isNone()) {
+            return QueryConstants.NULL_LONG;
+        }
+        return o.getLongValue();
+    }
+
+    public static float floatPyCast(Object a) {
+        if (a != null && !(a instanceof PyObject)) {
+            throw new IllegalArgumentException("provided value is not a PyObject");
+        }
+        PyObject o = (PyObject) a;
+        if (o == null || o.isNone()) {
+            return QueryConstants.NULL_FLOAT;
+        }
+        return (float) o.getDoubleValue();
+    }
+
+    public static char charPyCast(Object a) {
+        if (a != null && !(a instanceof PyObject)) {
+            throw new IllegalArgumentException("provided value is not a PyObject");
+        }
+        PyObject o = (PyObject) a;
+        if (o == null || o.isNone()) {
+            return QueryConstants.NULL_CHAR;
+        }
+        return (char) o.getIntValue();
+    }
+
+    public static byte bytePyCast(Object a) {
+        if (a != null && !(a instanceof PyObject)) {
+            throw new IllegalArgumentException("provided value is not a PyObject");
+        }
+        PyObject o = (PyObject) a;
+        if (o == null || o.isNone()) {
+            return QueryConstants.NULL_BYTE;
+        }
+        return (byte) o.getIntValue();
+    }
+
+    public static short shortPyCast(Object a) {
+        if (a != null && !(a instanceof PyObject)) {
+            throw new IllegalArgumentException("provided value is not a PyObject");
+        }
+        PyObject o = (PyObject) a;
+        if (o == null || o.isNone()) {
+            return QueryConstants.NULL_SHORT;
+        }
+        return (short) o.getIntValue();
+    }
+
+    public static String doStringPyCast(Object a) {
+        if (a == null) {
+            return null;
+        }
+        if (!(a instanceof PyObject)) {
+            throw new IllegalArgumentException("provided value is not a PyObject");
+        }
+        PyObject o = (PyObject) a;
+        if (o.isNone()) {
+            return null;
+        }
+        return o.getStringValue();
+    }
+
+    public static boolean booleanPyCast(Object a) {
```

Comment:
Ask Chip and Jianfeng about this. I think `null` and `isNone()` should probably NPE.